app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt

```diff
@@ -1,52 +1,135 @@
+            }
+            return false
+        } else if (secondOperandUtils.isEmpty()) {
+            error = ErrorUtils.ERROR_MESSAGE_INVALID_FORMAT
+            result = EMPTY_STRING
+            return false
         }
+        return true
+    }
+
+    override fun doOperations() {
+        if (isValidOperation()) {
+
+            when (operator) {
+                OPERATOR_SUM -> {
+                    result =
+                        (firstOperandUtils.getValue() + secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                OPERATOR_MINUS -> {
+                    result =
+                        (firstOperandUtils.getValue() - secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                OPERATOR_DIVIDE -> if (secondOperandUtils.getValue() == ZERO_NUMBER_DOUBLE_TYPE) {
+                    result = EMPTY_STRING
+                    error = ErrorUtils.ERROR_MESSAGE_DIVISION
+                } else {
+                    result =
+                        (firstOperandUtils.getValue() / secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                OPERATOR_MULTIPLY -> {
+                    result =
+                        (firstOperandUtils.getValue() * secondOperandUtils.getValue()).toString()
+                    error = ErrorUtils.NONE
+                }
+                else -> {
+                    ErrorUtils.ERROR_MESSAGE.toString()
+                    result = EMPTY_STRING
+                }
+
+            }
+            updateFirstOperand()
+            secondOperandUtils.eraseOperands()
+
+        }
+    }
+
+    override fun getResult(): String {
+        return result
+    }
+
+    private fun updateFirstOperand() {
+        if (result.substring(POSITION_ZERO).equals(OPERATOR_MINUS)) {
+            firstOperandUtils.signs = OPERATOR_MINUS
+            firstOperandUtils.value = result.substring(POSITION_ONE, result.length)
+        } else {
+            firstOperandUtils.signs = EMPTY_STRING
+            firstOperandUtils.value = result
+        }
+        operator = EMPTY_STRING
     }
 
     override fun eraseResult(): String {
         result = EMPTY_STRING
-        firstOperand = EMPTY_STRING
-        secondOperand = EMPTY_STRING
+        firstOperandUtils.eraseOperands()
+        secondOperandUtils.eraseOperands()
         operator = EMPTY_STRING
         return EMPTY_STRING
     }
 
     override fun saveOperationSymbol(operatorSymbol: String) {
-        operator = operatorSymbol
+        if (firstOperandUtils.isEmpty()) {
+            firstOperandUtils.setSign(OPERATOR_MINUS)
+        } else if (operator.isEmpty()) {
+            operator = operatorSymbol
+        } else {
+            secondOperandUtils.setSign(OPERATOR_MINUS)
+        }
+    }
+
+    override fun getError(): ErrorUtils {
+        return error
```

Comment:
same here you can do this     override fun getError() = error

Reply:
Thx for the tip, now its done